api/src/context.rs

```diff
@@ -277,6 +319,26 @@ impl Context {
         Ok((latest_ledger_info, requested_ledger_version))
     }
 
+    pub fn get_latest_internal_indexer_ledger_version_and_info<E: StdApiError>(
+        &self,
+    ) -> Result<(LedgerInfo, Version), E> {
+        if self.indexer_reader.is_none() {
+            return Err(E::internal_with_code_no_info(
+                "Indexer reader doesn't exist",
+                AptosErrorCode::InternalError,
+            ));
+        }
+        let latest_version = self
+            .indexer_reader
+            .as_ref()
+            .unwrap()
+            .get_latest_internal_indexer_ledger_version()
+            .map_err(|err| E::internal_with_code_no_info(err, AptosErrorCode::InternalError))?;
+        let mut latest_ledger_info = self.get_latest_ledger_info()?;
+        latest_ledger_info.ledger_version = aptos_api_types::U64(latest_version);
```

Comment:
This doesn't look right..  it can have a block timestamp or a block height inconsistent of the ledger version.  -- we should get those according to the ledger version.